src/main/java/nz/ac/auckland/kafka/http/sink/request/ApiRequestInvoker.java

```diff
@@ -58,7 +63,27 @@ public void invoke(final Collection<SinkRecord> records){
             MDC.put(ApiRequest.REQUEST_HEADER_INFO_KEY, kafkaDetails);
 
             log.info("Processing record: topic={}  partition={} offset={} value={}", record.topic(), record.kafkaPartition(), record.kafkaOffset(), record.value().toString());
-            sendAPiRequest(record , traceId);
+            
+            Long sleepTime = 0L;
+            if (record.timestampType().equals(TimestampType.NO_TIMESTAMP_TYPE)){
+                log.info("Checking delay: now={}, sleep={}", start,new Date(start).toString(),new Date(this.config.requestDelay*1000).toString());
+                // If there is no timestamp we sleep for the specified delay
+                sleepTime = this.config.requestDelay * 1000L;
+            } else { 
+                long timePassed = start - record.timestamp();
+                log.info("Checking delay: now={}, record-timestamp={}, timepassed={}, sleep={}", new Date(start).toString(),new Date(record.timestamp()).toString(),timePassed,(this.config.requestDelay*1000 - timePassed));
+                if (timePassed < this.config.requestDelay) {
+                    sleepTime = (this.config.requestDelay * 1000L) - timePassed;
+                }
+            }
```

Comment:
I'd be inclined to put this into a function i.e.
```suggestion
            Long messageDelay = getMessageDelay(record);
            
private int getMessageDelay(SinkRecord record) {
  Long delay = this.config.requestDelay * 1000L;

  if (record.timestampType().equals(TimestampType.NO_TIMESTAMP_TYPE)){
    log.info("Checking delay: now={}, sleep={}", start, new Date(start).toString(), new Date(delay).toString());
    // If there is no timestamp we sleep for the specified delay
    return delay;
  }

  Long timePassed = start - record.timestamp();
  Long delayTimePassed = delay - timePassed;

  log.info("Checking delay: now={}, record-timestamp={}, timepassed={}, sleep={}", new Date(start).toString(), new Date(record.timestamp()).toString(), timePassed, delayTimePassed);

  if (timePassed < this.config.requestDelay) {
    return delayTimePassed;
  }

  return 0L;
}
```